File: algorithmic_heights/merge_sort/merge_sort.hs
import System.IO.Unsafe
import Control.Monad
import Data.List.Split


mergeSort :: Ord a => [a] -> [a] -> [a]
mergeSort [] [] = []
mergeSort a [] = a
mergeSort [] a = a
mergeSort (x:xs) (y:ys) | x < y  = x : mergeSort xs (y:ys)
                        | y < x  = y : mergeSort (x:xs) ys
                        | y == x = y : x : mergeSort xs ys

readInt :: String -> Int
readInt = read

arrays = map (map (read :: String -> Int) . words) $ splitOn "\n" $ unsafePerformIO . readFile $ "rosalind_mer.txt"

main = do
        mapM print $ mergeSort (arrays !! 1) (arrays !! 3)
   

File: algorithmic_heights/merge_sort/merge_sort.py

def merge_sort(array1,array2):
    if array1 == []:
        return array2
    elif array2 == []:
        return array1
    elif array1[0] < array2:
        return array1[0] + merge_sort(array1[:1],array2)
    elif array1[0] > array2:
        return array2[0] + merge_sort(array1,array2[:1])
    elif array1[0] == array2[0]:
        return array1[0] + array2[0] + merge_sort(array1[:1],array2[:1])
    else:
        return 0
                

